### PR TITLE
Change dependency and build scripts to work on OS X

### PR DESCRIPTION
**Pull request by thilinarmtb**
I tested the changes on OSX Sierra (10.12.3) and works fine. It will be great if
someone can verify that I didn't break anything on linux side.

### CODE REVIEW

File: dependencies.sh [hunk 1]
```diff
@@ -1,11 +1,23 @@
...
-    wget http://downloads.dlang.org/releases/2.x/2.072.2/dmd.2.072.2.linux.zip
-    SUM2=`md5sum dmd.2.072.2.linux.zip`
+    if [[ "$OSTYPE" == "linux-gnu" ]]; then
+        FILE="dmd.2.072.2.linux.zip"
+        SUM1="c6dc5df0eeac35d37db167b9a80eb08e  $FILE"
+	ZIPLINK="http://downloads.dlang.org/releases/2.x/2.072.2/$FILE"
+	MD5="md5sum"
+    elif [[ "$OSTYPE" == "darwin"* ]]; then
+        FILE="dmd.2.072.2.osx.zip"
+	SUM1="MD5 (dmd.2.072.2.osx.zip) = 0844f3218043a21dcc7a3fc76605af5d"
```
Comment (thilinarmtb) [inline dependencies.sh]: I should replace `dmd.2.072.2.osx.zip` by $FILE.

File: dependencies.sh [hunk 2]
```diff
@@ -1,11 +1,23 @@
...
-    SUM1="c6dc5df0eeac35d37db167b9a80eb08e  dmd.2.072.2.linux.zip"
-    wget http://downloads.dlang.org/releases/2.x/2.072.2/dmd.2.072.2.linux.zip
-    SUM2=`md5sum dmd.2.072.2.linux.zip`
+    if [[ "$OSTYPE" == "linux-gnu" ]]; then
+        FILE="dmd.2.072.2.linux.zip"
+        SUM1="c6dc5df0eeac35d37db167b9a80eb08e  $FILE"
+	ZIPLINK="http://downloads.dlang.org/releases/2.x/2.072.2/$FILE"
+	MD5="md5sum"
+    elif [[ "$OSTYPE" == "darwin"* ]]; then
+        FILE="dmd.2.072.2.osx.zip"
```
Comment (thilinarmtb) [inline dependencies.sh]: Additional space at the beginning of line.

File: build.sh [hunk 3]
```diff
@@ -1,10 +1,12 @@
 #!/bin/bash
-if [ -d "dmd2" ]; then
```
Comment (tgehr) [inline build.sh:2]: This should stay in in some form (basically, if someone already has a compatible DMD compiler installed, I don't want to force them to run dependencies.sh.)

File: build.sh [hunk 4]
```diff
@@ -1,10 +1,12 @@
...
-    DMD="./dmd2/linux/bin64/dmd";
-else
-    DMD="dmd"
+if [[ "$OSTYPE" == "linux-gnu" ]]; then
+    BIN="linux/bin64"
+elif [[ "$OSTYPE" == "darwin"* ]]; then
+    BIN="osx/bin"
 fi
 
+DMD="./dmd2/$BIN/dmd"
```
Comment (tgehr) [inline build.sh]: You could add the if-statement above around this statement.

**Comment by tgehr** [comment]
Thanks! I left a comment, but other than that, everything is still running on the linux side.

**Comment by thilinarmtb** [comment]
Thanks for the comments !! I made the required changes.

**Comment by tgehr** [comment]
Thanks, merging.

**Comment by thilinarmtb** [comment]
Thanks for merging !!

**Pull request merged**